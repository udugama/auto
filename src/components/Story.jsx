import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Paper, Link } from '@material-ui/core';
import Typography from '@material-ui/core/Typography';
import leftPad from 'left-pad';

const useStyles = makeStyles(theme => ({
  root: {
    backgroundColor: '#f5f5f5', // '#333'
    padding: theme.spacing(0.5, 3),
    alignContent: leftPad,
    marginBottom: 12,
    marginLeft: 12,
    marginRight: 12,
    marginTop: 12,
  },
}));

export default function Story(props) {
  const classes = useStyles();
  const story = props.data;

  return (
    <div>
      <Paper className={classes.root} square={false} spacing={3}> 
      <Link href={story.url} color="inherit" target="_blank" rel="noreferrer" >
        <Typography component='p' align='left'>
            Title: {story.title}
        </Typography>
      </Link>
        <Typography variant='caption' component='h5' align='left' >
            Author: {story.author}
        </Typography>
      </Paper>
    </div>
  );
}