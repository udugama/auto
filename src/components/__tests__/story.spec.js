import React from 'react';
import ReactDOM from 'react-dom';
import Story from './../Story';
import { shallow } from 'enzyme';
import { data } from './../fixtures/data';
import { render } from '@testing-library/react';

const { hits } = data;

it('renders without crashing', () => {
    shallow(<Story data={hits[0]} />);
});

it('render stories', () => {
  const { getByText } = render(<Story data={hits[0]} />);
  expect(getByText).toMatchSnapshot();
});