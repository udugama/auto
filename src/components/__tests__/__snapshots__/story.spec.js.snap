// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render stories 1`] = `[Function]`;
